### PR TITLE
Fix TabBarBottom layout + Update docs

### PR DESCRIPTION
**Pull request by CharlesMangwa**
In the ß3, `<TabBarBottom />` was shipped with a little bug that leads to this behaviour: 

This PR aims to fix that bug 🙂

### CODE REVIEW

**Comment by LeoLeBras** [comment]
I'm quite busy currently by myself with lots of other stuff.  I'll merge it as soon as I get some time (next week).

**Comment by LeoLeBras** [comment]
```<DefaultRenderer />``` and ```<NavBar />``` doesn't work with new ```react-navigation``` version. It looks like you may need to make some changes to :

**```<DefaultRenderer />```:**
```diff
...
    return (
      <Card
        {...props}
        key={`card_${props.scene.key}`}
        panHandlers={panHandlers}
-       renderScene={() => this.renderInnerCard(props)}
        style={[
          style,
          this.props.cardStyle,
          ownProps && ownProps.cardStyle,
        ]}
-      />
+        {this.renderInnerCard(props)}
+      </Card>
    )
...
```
But the pan handlers no longer works. I will investigate the bug late.

**```<NavBar />```:**
You should pass a ```getScreenDetails``` prop with 3 option:
* `headerLeft`
* `headerTitle`
* `headerRight`

to fix the bug.

@CharlesMangwa Can you fix it ?

**Comment by CharlesMangwa** [comment]
@LeoLeBras Fix pushed in (318aa5a). Just need to work on NavBar and Pan Responder now, and we should be good 🙌

**Comment by CharlesMangwa** [comment]
@LeoLeBras I removed `react-navigation` update to this PR as we said. I don't know if you plan to push some majors changes to the API, because I'm thinking about doing something cool for the doc 😏

File: modules/DefaultRenderer.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import React, { Component } from 'react'
 import { NativeModules, StyleSheet, Platform, View } from 'react-native'
 import { Transitioner } from 'react-navigation'
 import Card from 'react-navigation/src/views/Card'
-import CardStackPanResponder from 'react-navigation/src/views/CardStackPanResponder'
+import CardStackStyleInterpolator from 'react-navigation/src/views/CardStackStyleInterpolator'
```
Comment (LeoLeBras) [inline modules/DefaultRenderer.js]: Please, remove this changes. ```react-navigation@1.0.0-beta.8``` will be support in [next branch](https://github.com/LeoLeBras/react-router-navigation/tree/next)

Comment (CharlesMangwa) [inline modules/DefaultRenderer.js]: Hòla! Fixed in https://github.com/LeoLeBras/react-router-navigation/pull/14/commits/34b92d9b53c28d9722f01a9c9c3c24fbc9bdac52

File: jsconfig.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (LeoLeBras) [inline jsconfig.json]: Please, remove this file !

File: modules/TabStack.js [hunk 3]
```diff
@@ -113,26 +113,38 @@ class TabStack extends Component<DefaultProps, OwnProps, State> {
   onRequestChangeTab = (index: number): void => {
     const entries = this.state.history[index]
     if (index !== this.state.navigationState.index) {
```
Comment (LeoLeBras) [inline modules/TabStack.js]: Why do you add this ?

File: modules/TypeDefinitions.js [hunk 4]
```diff
@@ -20,7 +20,11 @@ export type RouteProps = {
   children?: (props: ContextRouter) => React$Element<any>,
   path: string,
   exact?: boolean,
-  strict?: boolean
+  strict?: boolean,
+  state?: NavigationState<{
```
Comment (LeoLeBras) [inline modules/TypeDefinitions.js]: Hum ... Why this change ?

Comment (CharlesMangwa) [inline modules/TypeDefinitions.js]: Because of #19 :)

File: modules/TabStack.js (hunk 3 above)
Comment (CharlesMangwa) [inline modules/TabStack.js]: Because of #19 IIRC :)

File: jsconfig.json (hunk 2 above)
Comment (CharlesMangwa) [inline jsconfig.json]: My bad, VS Code playing around with me ^^

File: modules/TabStack.js (hunk 3 above)
Comment (LeoLeBras) [inline modules/TabStack.js]: Could you please create a new pr ? 😎

Comment (CharlesMangwa) [inline modules/TabStack.js]: Is that really necessary ^^'?… I could if you insist, but that would be to cumbersome imho (why not just change the PR title haha?)

File: modules/TypeDefinitions.js (hunk 4 above)
Comment (LeoLeBras) [inline modules/TypeDefinitions.js]: But this is wrong because `state` must just be an `Object` not a `NavigationState`

Comment (CharlesMangwa) [inline modules/TypeDefinitions.js]: OK! I was about to use `Object`in the 1st place, but it sound pretty awkward to me

Comment (LeoLeBras) [inline modules/TypeDefinitions.js]: Why ?

Comment (CharlesMangwa) [inline modules/TypeDefinitions.js]: I can't put my finger on it… Anyway: fixed!

File: modules/TabStack.js (hunk 3 above)
Comment (CharlesMangwa) [inline modules/TabStack.js]: 😨

File: docs/BOTTOM_NAVIGATION.md [hunk 5]
```diff
@@ -25,22 +24,22 @@ const App = () => (
...
 * **tabActiveTintColor** ```?string``` label and icon color of the active tab
 * **label** ```?string``` text that appears on each item
 * **labelStyle** ```?(StyleSheet | Function)``` styling text item
 * **renderLabel** ```?Function``` callback which renders a label
-* **renderTabIcon** ```?Function``` optional callback which receives the current scene and returns a React Element to be used as a icon
+* **renderTabIcon** ```?Function``` optional callback which receives the current scene and returns a React Element to be used as an icon
 
 #### ```<BottomNavigation />``` Props
-* [```... TarBar Props```](https://github.com/LeoLeBras/react-router-navigation/blob/master/docs/BOTTOM_NAVIGATION.md#tabbar-props)
+* [```... TabBar props```](https://github.com/LeoLeBras/react-router-navigation/blob/master/docs/BOTTOM_NAVIGATION.md#tabbar-props)
```
Comment (LeoLeBras) [inline docs/BOTTOM_NAVIGATION.md]: Can you make the same change at line `25`

File: docs/NAVIGATION.md [hunk 6]
```diff
@@ -42,11 +42,11 @@ const App = () => (
 * **renderRightButton** ```?Function``` callback which renders a right button
 
 #### ```<Navigation />``` Props
-* [... NavBar Props](https://github.com/LeoLeBras/react-router-navigation/blob/master/docs/NAVIGATION.md#navbar-props)
+* [... ```<NavBar />```  props](https://github.com/LeoLeBras/react-router-navigation/blob/master/docs/NAVIGATION.md#navbar-props)
```
Comment (LeoLeBras) [inline docs/NAVIGATION.md]: Can you make the same change at line `31` ?

File: docs/TABS.md [hunk 7]
```diff
@@ -37,12 +37,12 @@ const App = () => (
 * **renderLabel** ```?Function``` callback which renders a label
 
 #### ```<Tabs />``` Props
-* [```... TarBar Props```](https://github.com/LeoLeBras/react-router-navigation/blob/master/docs/BOTTOM_NAVIGATION.md#tabbar-props)
+* [```... TabBar props``` props](https://github.com/LeoLeBras/react-router-navigation/blob/master/docs/BOTTOM_NAVIGATION.md#tabbar-props)
```
Comment (LeoLeBras) [inline docs/TABS.md]: Can you make the same change at line `26` ?

File: docs/BOTTOM_NAVIGATION.md (hunk 5 above)
Comment (CharlesMangwa) [inline docs/BOTTOM_NAVIGATION.md]: Done.

File: docs/NAVIGATION.md (hunk 6 above)
Comment (CharlesMangwa) [inline docs/NAVIGATION.md]: Done.

File: docs/TABS.md (hunk 7 above)
Comment (CharlesMangwa) [inline docs/TABS.md]: And done.
